File: frontend/src/services/adminUserService.js
import axiosInstance from '../config/axios';

export const adminUserService = {
  getUsersByRole: async (role) => {
    const response = await axiosInstance.get(`/api/admin/users?role=${role}`);
    return response.data;
  },
  
  getUserById: async (id) => {
    const response = await axiosInstance.get(`/api/admin/users/${id}`);
    return response.data;
  },

  createUser: async (userData) => {
    const response = await axiosInstance.post('/api/admin/users', userData);
    return response.data;
  },

  updateUser: async (id, userData) => {
    const response = await axiosInstance.put(`/api/admin/users/${id}`, userData);
    return response.data;
  },

  lockUser: async (id) => {
    const response = await axiosInstance.delete(`/api/admin/users/${id}`);
    return response.data;
  },

  unlockUser: async (id) => {
    const response = await axiosInstance.post(`/api/admin/users/${id}/unlock`);
    return response.data;
  }
};

File: backend/src/main/java/com/project/event/service/AdminUserService.java
package com.project.event.service;

import com.project.event.dto.AdminUserRequest;
import com.project.event.dto.AdminUserResponse;

import java.util.List;
import java.util.UUID;

public interface AdminUserService {
    List<AdminUserResponse> getUsersByRole(String role);
    AdminUserResponse getUserById(UUID id);
    AdminUserResponse createUser(AdminUserRequest request);
    AdminUserResponse updateUser(UUID id, AdminUserRequest request);
    void lockUser(UUID id);
    void unlockUser(UUID id);
}
